src/types.d.ts

```diff
@@ -6,7 +8,9 @@ interface ProgressStatus {
 
 export interface ProgressFn { (status: ProgressStatus): void }
 
-export interface FetchOptions extends RequestInit {
+type Modify<T, R> = Omit<T, keyof R> & R
```

Comment:
I think that would be a more descriptive name

```suggestion
type Override<T, R> = Omit<T, keyof R> & R
```